frontend/src/main/java/pl/pollub/frontend/service/AdminService.java

```diff
@@ -0,0 +1,174 @@
+                throw new RuntimeException("Błąd pobierania użytkowników: " + response.statusCode());
+            } else {
+                List<User> userList = new ArrayList<>();
+                try {
+                    String responseBody = response.body();
+                    Map<String, Object>[] usersArray = JsonUtil.GSON.fromJson(responseBody, Map[].class);
+
+                    for (Map<String, Object> userMap : usersArray) {
+                        User user = new User();
+
+                        Object idObj = userMap.get("id");
+                        if (idObj != null) {
+                            user.setId(((Number) idObj).longValue());
+                        } else {
+                            user.setId(0);
+                        }
+
+                        user.setUsername((String) userMap.get("username"));
+                        user.setEmail((String) userMap.get("email"));
+
+                        String role = (String) userMap.get("role");
+                        user.setRole(role);
+
+                        user.setAdmin("ADMIN".equals(role));
+
+                        Object monthlyLimitObj = userMap.get("monthlyLimit");
+                        if (monthlyLimitObj != null) {
+                            user.setMonthlyLimit(((Number) monthlyLimitObj).doubleValue());
+                        } else {
+                            user.setMonthlyLimit(null);
+                        }
+
+                        userList.add(user);
+                    }
+                    return userList;
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    throw new RuntimeException("Błąd przy konwersji JSON do listy użytkowników");
+                }
+            }
+        }
+
+    public HttpResponse<String> updateEmail(Long userId, String email) {
+        Map<String, String> profileData = new HashMap<>();
+        profileData.put("email", email);
+
+        String json = JsonUtil.GSON.toJson(profileData);
+        return httpService.put("/api/admin/" + userId + "/email", json);
+    }
+
+    public HttpResponse<String> updateUsername(Long userId, String username) {
+        Map<String, String> profileData = new HashMap<>();
+        profileData.put("username", username);
+        String json = JsonUtil.GSON.toJson(profileData);
+        return httpService.put("/api/admin/" + userId + "/username", json);
+    }
+
+    public HttpResponse<String> updateLimit(Long userId, Double limit) {
+        Map<String, Object> profileData = new HashMap<>();
+        profileData.put("spendingLimit", limit);
+        String json = JsonUtil.GSON.toJson(profileData);
+        return httpService.put("/api/admin/" + userId + "/limit", json);
+    }
+
+    public HttpResponse<String> updateRole(Long userId, String role) {
+        Map<String, String> profileData = new HashMap<>();
+        profileData.put("role", role);
+        String json = JsonUtil.GSON.toJson(profileData);
```

Comment:
here and above you dont have to manually stringify this to json, you can pass a map object into the http service methods